src/EditorFeatures/Test2/IntelliSense/IntellisenseQuickInfoBuilderTests.vb

```diff
@@ -12,88 +17,476 @@ Imports QuickInfoItem = Microsoft.CodeAnalysis.QuickInfo.QuickInfoItem
+
+                Dim languageServiceProvider = workspace.Services.GetLanguageServices(language)
+                Dim quickInfoService = languageServiceProvider.GetRequiredService(Of QuickInfoService)
+
+                Return Await quickInfoService.GetQuickInfoAsync(document, cursorPosition, CancellationToken.None).ConfigureAwait(False)
+            End Using
+        End Function
+
+        Private Shared Sub AssertEqualAdornments(expected As Object, actual As Object)
+            Assert.IsType(expected.GetType, actual)
+
+            Dim containerElement = TryCast(expected, ContainerElement)
+            If containerElement IsNot Nothing Then
+                AssertEqualContainerElement(containerElement, DirectCast(actual, ContainerElement))
+                Return
+            End If
+
+            Dim imageElement = TryCast(expected, ImageElement)
+            If imageElement IsNot Nothing Then
+                AssertEqualImageElement(imageElement, DirectCast(actual, ImageElement))
+                Return
+            End If
+
+            Dim classifiedTextElement = TryCast(expected, ClassifiedTextElement)
+            If classifiedTextElement IsNot Nothing Then
+                AssertEqualClassifiedTextElement(classifiedTextElement, DirectCast(actual, ClassifiedTextElement))
+                Return
+            End If
+
+            Dim classifiedTextRun = TryCast(expected, ClassifiedTextRun)
+            If classifiedTextRun IsNot Nothing Then
+                AssertEqualClassifiedTextRun(classifiedTextRun, DirectCast(actual, ClassifiedTextRun))
+                Return
+            End If
+
+            Throw Roslyn.Utilities.ExceptionUtilities.Unreachable
+        End Sub
+
+        Private Shared Sub AssertEqualContainerElement(expected As ContainerElement, actual As ContainerElement)
+            Assert.Equal(expected.Style, actual.Style)
+            Assert.Equal(expected.Elements.Count, actual.Elements.Count)
+            For Each pair In expected.Elements.Zip(actual.Elements, Function(expectedElement, actualElement) (expectedElement, actualElement))
+                AssertEqualAdornments(pair.expectedElement, pair.actualElement)
+            Next
+        End Sub
+
+        Private Shared Sub AssertEqualImageElement(expected As ImageElement, actual As ImageElement)
+            Assert.Equal(expected.ImageId.Guid, actual.ImageId.Guid)
+            Assert.Equal(expected.ImageId.Id, actual.ImageId.Id)
+            Assert.Equal(expected.AutomationName, actual.AutomationName)
+        End Sub
+
+        Private Shared Sub AssertEqualClassifiedTextElement(expected As ClassifiedTextElement, actual As ClassifiedTextElement)
+            Assert.Equal(expected.Runs.Count, actual.Runs.Count)
+            For Each pair In expected.Runs.Zip(actual.Runs, Function(expectedRun, actualRun) (expectedRun, actualRun))
+                AssertEqualClassifiedTextRun(pair.expectedRun, pair.actualRun)
+            Next
+        End Sub
+
```

Comment:
the test seems missing some cases for state machine for GetClassifiedText method. like line breaks at the start or 2+ line breaks and etc.

Reply:
Tests have been added 👍